### PR TITLE
Expose the symbol provider of a `SymbolOccurrence` via the Swift API

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/IndexStoreDB/IndexStoreDB.swift [hunk 1]
```diff
@@ -289,16 +302,7 @@ public final class IndexStoreDB {
   public func symbolProvider(for sourceFilePath: String) -> SymbolProviderKind? {
```
Comment (bnbarham) [inline Sources/IndexStoreDB/IndexStoreDB.swift:302]: IIRC this API was added for what's now using the provider from the occurrence instead, should we just remove?

**Review by bnbarham — APPROVED** [review]

File: Sources/IndexStoreDB/IndexStoreDB.swift (hunk 1 above)
Comment (bnbarham) [reply Sources/IndexStoreDB/IndexStoreDB.swift:302]: (https://github.com/apple/indexstore-db/pull/177)

Comment (ahoppen) [reply Sources/IndexStoreDB/IndexStoreDB.swift:302]: I’ll remove it in a follow-up PR.

**Pull request merged**